graph/adjacency/impl/ThingAdjacencyImpl.java

```diff
@@ -387,66 +302,60 @@ private ThingEdgeImpl put(Encoding.Edge.Thing encoding, ThingEdgeImpl edge, IID[
                 );
             }
 
-            edges.compute(infixIID, (iid, cachedEdges) -> {
-                if (cachedEdges == null) cachedEdges = new ConcurrentSkipListMap<>();
-                ComparableEdge.Thing comparableEdge = toComparable(edge);
-                ThingEdge thingEdge = cachedEdges.get(comparableEdge);
-                if (thingEdge != null) {
-                    if (thingEdge.isInferred() && !edge.isInferred()) thingEdge.isInferred(false);
+            edges.compute(infixIID, (iid, bufferedEdges) -> {
+                EDGE_VIEW edgeView = getView(edge);
+                if (bufferedEdges == null) {
+                    bufferedEdges = new ConcurrentSkipListMap<>();
+                    bufferedEdges.put(edgeView, edge);
                 } else {
-                    cachedEdges.put(comparableEdge, comparableEdge.edge());
+                    ThingEdge existingEdge = bufferedEdges.get(edgeView);
+                    if (existingEdge == null) bufferedEdges.put(edgeView, edge);
+                    else if (existingEdge.isInferred() && !edge.isInferred()) existingEdge.isInferred(false);
```

Comment:
refactor the adding or updating a buffered edge to make the flow a bit clearer, and also reduce eliminate one `get` when the buffered edges are just being initialised